cmd/helm-migrate-values/root.go

```diff
@@ -60,14 +60,16 @@ func NewRootCmd(actionConfig *action.Configuration, settings *cli.EnvSettings, o
 	var outputFile string
 	flags.StringVarP(&outputFile, "output-file", "o", "",
 		"The output file to which the result is saved. Standard output is used if this option is not set.")
+	var migrationDir string
+	flags.StringVar(&migrationDir, "migration-dir", "value-migrations", "Specifies the relative path to the directory containing migration definition files. The path should be relative to the Helm chart directory.")
 
-	runner := newRunner(actionConfig, flags, settings, out, &outputFile, log)
+	runner := newRunner(actionConfig, flags, settings, out, &migrationDir, &outputFile, log)
 	cmd.RunE = runner
 
 	return cmd, nil
 }
 
-func newRunner(actionConfig *action.Configuration, flags *pflag.FlagSet, settings *cli.EnvSettings, out io.Writer, outputFile *string, log pkg.Logger) func(cmd *cobra.Command, args []string) error {
+func newRunner(actionConfig *action.Configuration, flags *pflag.FlagSet, settings *cli.EnvSettings, out io.Writer, migrationDir *string, outputFile *string, log pkg.Logger) func(cmd *cobra.Command, args []string) error {
```

Comment:
There's a tiny performance/memory benefit to passing `migrationDir` as a reference here, but I think we should pass it as a value to indicate that it can/should never be `nil` at this point.

Reply:
It needs to be a pointer here because the flags aren't parsed until after the setup code has run. So if this were a normal string, then the value stored in `var migrationDir string` at the time the `newRunner` is called will be passed by value i.e. you'll always get the default value of the flag.